0018-connector-risk-mitigations/0018-connector-risk-mitigations.md

```diff
@@ -16,29 +16,53 @@ The main risk connectors face in Interledger is being unable to fulfill the inco
 
 Possible mitigations include:
 
 * **Managing "Payment Bandwidth"** - Connectors should monitor the percentage of their total liquidity that is on hold at any given time for a given peer, customer, or destination prefix and may reject incoming payment requests if that party exceeds their allocated "payment bandwidth". Connectors may allocate less bandwidth to unknown or untrustworthy senders or receivers. Note that ILP Packets contain the destination account but not the source account, so connectors should apply this logic to the immediate peer or customer that they receive the payments from and possibly the destination account or prefix.
-* **Preferring Smaller Payments** - Smaller payments place a smaller percentage of the connector's liquidity on hold and thus each one carries less risk that the payment being prepared and then failing would create significant opportunity cost
-* **Blacklisting Senders and Receivers** - Connectors may refuse to facilitate payments from certain sources or to certain destinations if their rate of failed payments is unusually high
+* **Preferring Smaller Payments** - Smaller payments place a smaller percentage of the connector's liquidity on hold and thus each one carries less risk that the payment being prepared and then failing would create significant opportunity cost - in ILPv4, this is the default.
+* **Blacklisting Senders and Receivers** - Connectors may refuse to facilitate payments from certain sources or to certain destinations if their rate of failed payments is unusually high.
+
+Counterparty Theft: Suppose you have connectors A, B, C and D, and the payment is going from A->B->C->D. Take the case of C. During the fulfill phase, D will pass a fulfill packet to C, meaning that C will have to pay. But because ILP is using unconditional payment channels, C can write a malicious client that could forward a fulfill packet to B, and avoid paying D. In this case, A would pay B, B would pay C, but C didn't pay D.
+
+Possible mitigations include:
+* **Accept only smaller payments** - If the payment per packet is less than that
+    of an on chain-fee, a connector that cheats it counterparty would lose much
+    more capital in the process. For example, a connector that has been deprived of
+    payment from its counterparty could refuse to facilitate future payments,
+    and the malicious counterparty would have to close their payment channel to
+    make the stolen micropayment amount liquid. Given that on-chain transaction fees
+    are much higher than that of a micropayment in Interledger (i.e. 1 satoshi), an
+    attacker would lose more money open and closing channels to cheat their
+    counterparty. Therefore, while this griefiing attack is possible, if
+    connectors are leveraging smaller payments, the attacker is more incentivized
+    to facilitate payments to make profit on the exchange rate/fees.
+* **Leverage payment channels with fraud proofs** - (TODO: Wondering if this is possible?)
```

Comment:
Wondering if there was any insights into leveraging payment channels with fraud proofs to punish griefers/prevent them from continuing. i.e. if someone were to be slashed on the basis of this, that would be pretty cool.

This would basically mean that ILP's construction would start to bleed more into payment channels and ledgers would have to support this kind of proof (likely only Ethereum because of turing complete smart contract language).

Reply:
One perspective I would change throughout this document is that connectors are incentivized to continue participating in the network because they will take a margin on packets that pass through.  That is only one of the possible reasons for connectors to continue participating.  

You can imagine a relationship where a connector has a business model similar to an ISP where they charge their customers a flat rate for a certain amount of bandwidth and 0% margins.  In this scenario connectors are not incentivized to stay connected to the greater network to earn some percentage profit, but instead to provide a good customer experience where all of their customers can communicate value to any other entity in the greater network.

Reply:
Ah I'm seeing what you're getting at now. I misread what you initially said.
I just pushed something that should add some details regarding this.